### PR TITLE
Add io.github.FedericoCalzoni.EventRecorder

### PR DESCRIPTION
**Pull request by FedericoCalzoni**
EventRecorder is a simple application to record timed events. It is useful to keep track of the time spent on different tasks, and to generate reports.

### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: io.github.FedericoCalzoni.EventRecorder.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            ],
+            "build-commands": [
+                "install -D EventRecorder/EventRecorder/EventRecorder.py /app/bin/EventRecorder.py",
+                "chmod +x /app/bin/EventRecorder.py",
+                "install -D EventRecorder/EventRecorder/logo.png /app/share/icons/hicolor/128x128/apps/io.github.FedericoCalzoni.EventRecorder.png",
+                "install -D EventRecorder/io.github.FedericoCalzoni.EventRecorder.desktop /app/share/applications/io.github.FedericoCalzoni.EventRecorder.desktop"
+            ]
+        }
+    ],
+    "finish-args": [
```
Comment (hfiguiere) [inline io.github.FedericoCalzoni.EventRecorder.json:32]: this should be above modules.

File: io.github.FedericoCalzoni.EventRecorder.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        "--socket=fallback-x11",
+        "--share=ipc",
+        "--socket=wayland",
+        "--filesystem=home"
```
Comment (hfiguiere) [inline io.github.FedericoCalzoni.EventRecorder.json:36]: why do you need a broad filesystem permission?

File: io.github.FedericoCalzoni.EventRecorder.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://github.com/FedericoCalzoni/EventRecorder/archive/refs/tags/v1.0.0.tar.gz",
+                    "dest": "EventRecorder",
+                    "sha256": "a324e76289859a8e26c4a5e9088880ea676578e1f773afffbb8314f001018b2a"
+                }
+            ],
+            "build-commands": [
+                "install -D EventRecorder/EventRecorder/EventRecorder.py /app/bin/EventRecorder.py",
+                "chmod +x /app/bin/EventRecorder.py",
```
Comment (hfiguiere) [inline io.github.FedericoCalzoni.EventRecorder.json:26]:
```suggestion
                "install -Dm755 EventRecorder/EventRecorder/EventRecorder.py /app/bin/EventRecorder.py",
```

File: io.github.FedericoCalzoni.EventRecorder.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                "install -D EventRecorder/EventRecorder/logo.png /app/share/icons/hicolor/128x128/apps/io.github.FedericoCalzoni.EventRecorder.png",
+                "install -D EventRecorder/io.github.FedericoCalzoni.EventRecorder.desktop /app/share/applications/io.github.FedericoCalzoni.EventRecorder.desktop"
```
Comment (hfiguiere) [inline io.github.FedericoCalzoni.EventRecorder.json:28]: it's missing the mandatory appstream file.

File: io.github.FedericoCalzoni.EventRecorder.json (hunk 1 above)
Comment (FedericoCalzoni) [reply io.github.FedericoCalzoni.EventRecorder.json]: it should be fixed in the commit I have just made

File: io.github.FedericoCalzoni.EventRecorder.json (hunk 2 above)
Comment (FedericoCalzoni) [reply io.github.FedericoCalzoni.EventRecorder.json]: I have reduced the filesystem permission to only include: 
"--filesystem=~/.config/EventRecorder.ini"

The app works, but the shown path of the opened file is not the one of the filesystem but rather:
/run/user/1000/doc/982531fc/prova1.csv

I don't know if there is a way to show the proper one (at least what the user expect to be).

File: io.github.FedericoCalzoni.EventRecorder.json (hunk 3 above)
Comment (FedericoCalzoni) [reply io.github.FedericoCalzoni.EventRecorder.json]: I included it in the commit I made

File: io.github.FedericoCalzoni.EventRecorder.json (hunk 2 above)
Comment (bbhtt) [reply io.github.FedericoCalzoni.EventRecorder.json]: >  "--filesystem=~/.config/EventRecorder.ini"

Are you storing configs relative to HOME or XDG_CONFIG_HOME? The latter is the correct way and does not need any filesystem permission

**Comment by bbhtt** [comment]
bot, build io.github.FedericoCalzoni.EventRecorder

File: io.github.FedericoCalzoni.EventRecorder.json (hunk 2 above)
Comment (hfiguiere) [reply io.github.FedericoCalzoni.EventRecorder.json]: otherwise `--persist` is a better alternative.

Comment (FedericoCalzoni) [reply io.github.FedericoCalzoni.EventRecorder.json]: I made a new version supporting the XDG directories, and updated the manifest for the new version.

**Comment by FedericoCalzoni** [comment]
bot, build io.github.FedericoCalzoni.EventRecorder

**Comment by FedericoCalzoni** [comment]
bot, build io.github.FedericoCalzoni.EventRecorder

**Comment by FedericoCalzoni** [comment]
bot, build io.github.FedericoCalzoni.EventRecorder

File: io.github.FedericoCalzoni.EventRecorder.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "base": "com.riverbankcomputing.PyQt.BaseApp",
+    "base-version": "6.7",
+    "cleanup-commands": [
+        "/app/cleanup-BaseApp.sh"
+    ],
+    "command": "EventRecorder.py",
+    "finish-args": [
+        "--socket=fallback-x11",
+        "--share=ipc",
+        "--socket=wayland",
```
Comment (bbhtt) [inline io.github.FedericoCalzoni.EventRecorder.json:15]:
```suggestion
        "--socket=wayland",
        "--device=dri",
```

image

**Comment by bbhtt** [comment]
Is saving the CVS file working? I tried saving it in downloads folder but nothing seems to happen.

**Comment by FedericoCalzoni** [comment]
> Is saving the CVS file working? I tried saving it in downloads folder but nothing seems to happen.

You are right, saving the CSV it's not working. (The issue is limited to the flatpak build without --filesystem=home" permission.
Loading is working instead (if a valid CSV is already present, it can modify it properly).
I'm looking into it.

**Comment by FedericoCalzoni** [comment]
bot, build io.github.FedericoCalzoni.EventRecorder

**Comment by bbhtt** [comment]
Is the issue fixed now?

**Comment by FedericoCalzoni** [comment]
> Is the issue fixed now?

yes, it is

**Comment by bbhtt** [comment]
bot, build io.github.FedericoCalzoni.EventRecorder

**Pull request closed without merging**